plugin/helpers_bots.go

```diff
@@ -72,3 +113,67 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, botUserId string, options ...ShouldProcessMessageOption) bool {
```

Comment:
I don't like the required `botUserId` parameter. There are cases where you don't need to provide it, like just wanting to skip all system messages. What do you think about adding `botUserId` as a parameter to `AllowSelf()`?

Reply:
Point taken. Just trying to understand. The only problem occurs when I don't pass `AllowSelf()` then the `botUserID` won't be available for comparison.

Reply:
You only need `botUserID` for `AllowSelf` right? See also by comment https://github.com/mattermost/mattermost-server/pull/12401#discussion_r329481415

Reply:
Even if we remove the `IsBotDMChannel` check, if I don't pass the `AllowSelf()` the `AllowSelf` variable remains false and the check goes to https://github.com/RajatVaryani/mattermost-server/blob/ac616eaeab10723c0f59b8bc3cea8ea840a435c6/plugin/helpers_bots.go#L123 Here the value of botUserId is needed.

Reply:
Hm, I'm wondering if `AllowSelf` is needed at all. https://github.com/RajatVaryani/mattermost-server/blob/ac616eaeab10723c0f59b8bc3cea8ea840a435c6/plugin/helpers_bots.go#L155-L159 will filter out any bot posts anyway. Do you see value in this check? Is there a case where you want to skip bot posts, but not your own bot?

Reply:
@hanzei Logically it makes sense. I will remove the `AllowSelf`.

Reply:
Removed Allowself.